spec/specification.yaml

```diff
@@ -0,0 +1,204 @@
+openapi: 3.0.2
```

Comment:
See my comments on the previous PR related to this file.

Reply:
- [x] Typos
- [x] Pagination similar to QueryResponse in another project
- [x] Fixed urls